controllers/managedocs_controller_test.go

```diff
@@ -2,639 +2,759 @@ package controllers
-				ctx := context.Background()
-
-				// Verify strict mode
-				managedOCS := &v1.ManagedOCS{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      ManagedOCSName,
-						Namespace: TestNamespace,
-					},
-				}
+		When("all prometheus replicas are available", func() {
+			It("should reflect that in the ManagedOCS resource status", func() {
+				By("by seeting Status.Components.Prometheus.State to Ready")
+				// Ensure that the prometheus resource has unavailable replicas
+				prom := promTemplate.DeepCopy()
+				Expect(k8sClient.Get(ctx, utils.GetResourceKey(prom), prom)).Should(Succeed())
+
+				// Updating the status of the prometheus resource should trigger a reconcile for managedocs
+				prom.Status.UnavailableReplicas = 0
+				Expect(k8sClient.Update(ctx, prom)).Should(Succeed())
+
+				managedOCS := managedOCSTemplate.DeepCopy()
+				key := utils.GetResourceKey(managedOCS)
+				Eventually(func() v1.ComponentState {
+					Expect(k8sClient.Get(ctx, key, managedOCS)).Should(Succeed())
+					return managedOCS.Status.Components.Prometheus.State
+				}, timeout, interval).Should(Equal(v1.ComponentReady))
+			})
+		})
+		When("alertmanager has unavailable replicas", func() {
+			It("should reflect that in the ManagedOCS resource status", func() {
+				By("by seeting Status.Components.Alertmanager.State to Pending")
+				// Ensure that the prometheus resource has unavailable replicas
+				am := amTemplate.DeepCopy()
+				Expect(k8sClient.Get(ctx, utils.GetResourceKey(am), am)).Should(Succeed())
+
+				// Updating the status of the prometheus resource should trigger a reconcile for managedocs
+				am.Status.UnavailableReplicas = 1
+				Expect(k8sClient.Update(ctx, am)).Should(Succeed())
+
+				managedOCS := managedOCSTemplate.DeepCopy()
+				key := utils.GetResourceKey(managedOCS)
+				Eventually(func() v1.ComponentState {
+					Expect(k8sClient.Get(ctx, key, managedOCS)).Should(Succeed())
+					return managedOCS.Status.Components.Alertmanager.State
+				}, timeout, interval).Should(Equal(v1.ComponentPending))
+			})
+		})
+		When("all alertmanager replicas are available", func() {
+			It("should reflect that in the ManagedOCS resource status", func() {
+				By("by seeting Status.Components.Alertmanager.State to Ready")
+				// Ensure that the prometheus resource has unavailable replicas
+				am := amTemplate.DeepCopy()
+				Expect(k8sClient.Get(ctx, utils.GetResourceKey(am), am)).Should(Succeed())
+
+				// Updating the status of the prometheus resource should trigger a reconcile for managedocs
+				am.Status.UnavailableReplicas = 0
+				Expect(k8sClient.Update(ctx, am)).Should(Succeed())
+
+				managedOCS := managedOCSTemplate.DeepCopy()
+				key := utils.GetResourceKey(managedOCS)
+				Eventually(func() v1.ComponentState {
+					Expect(k8sClient.Get(ctx, key, managedOCS)).Should(Succeed())
+					return managedOCS.Status.Components.Alertmanager.State
+				}, timeout, interval).Should(Equal(v1.ComponentPending))
```

Comment:
```suggestion
				}, timeout, interval).Should(Equal(v1.ComponentReady))
```
This should be checking for ready.

Reply:
Nice catch